src/lib.rs

```diff
@@ -5687,7 +5691,55 @@ pub unsafe trait SplitByteSlice: ByteSlice {
     ///
     /// `x.split_at(mid)` panics if `mid > x.deref().len()`.
     #[must_use]
-    fn split_at(self, mid: usize) -> (Self, Self);
+    #[inline]
+    fn split_at(self, mid: usize) -> (Self, Self) {
+        if let Some(splits) = try_split_at(self, mid) {
+            splits
+        } else {
+            panic!("mid > len")
+        }
+    }
+
+    /// Splits the slice at the midpoint, possibly omitting bounds checks.
+    ///
+    /// `x.split_at_unchecked(mid)` returns `x[..mid]` and `x[mid..]`.
+    ///
+    /// # Safety
+    ///
+    /// `mid` must not be greater than `x.deref().len()`.
+    ///
+    /// # Panics
+    ///
+    /// Some implementations may panic instead of exhibiting undefined
+    /// behavior when `mid` is greater than `x.deref().len()`.
```

Comment:
I'm going back on this. If you violate a method's safety preconditions, then *anything* could happen. Specifying that a panic in particular is a subset of *anything* is redundant.

Reply:
Done.